assets/js/rootLantiq.js

```diff
@@ -2,40 +2,80 @@ const LOAD_ADDR = "80800000"
 const IMAGE0_ADDR = "C0000 740000";
 const IMAGE1_ADDR = "800000 800000";
 
-async function detectUboot(reader) {
+const textDecoder = new TextDecoder();
+const textEncoder = new TextEncoder();
+
+async function detectUboot(port) {
+    let reader = undefined;
+
     while (true) {
-        const { value, done } = await reader.read();
+        try {
+            if (reader == undefined) {
+                reader = port.readable.getReader();
+            }
+
+            const { value, done } = await reader.read();
+            const linesValue = textDecoder.decode(value).split('\n');
```

Comment:
I am not sure if this always works, I remember that I had put the LineBreakTransformer precisely because there was a risk that the sentence could be split over two different transmissions

we should keep the last line of the previous chunk and hang it

Reply:
I have no idea how to do this

Reply:
perhaps it does not make sense, but the one I suggested is an implementation that is used for instance in apache lucene filters